gluten-core/src/main/java/org/apache/gluten/vectorized/GeneralOutIterator.java

```diff
@@ -60,10 +61,18 @@ public final void close() {
     }
   }
 
+  public final void cancel() {
```

Comment:
@rui-mo I added a new interface for GeneralOutIterator. For velox it is by default empty. It is needed by clickhouse backend to cancel the pipeline before task is killed, to ensure some resources are released gracefully. Could you help review the change? Thanks!

Reply:
Is it possible to place the API directly to [CH's out iterator impl](https://github.com/apache/incubator-gluten/blob/e1469f0c3859c9dd5f0edb9d2f5890794a540cee/backends-clickhouse/src/main/java/org/apache/gluten/vectorized/BatchIterator.java#L27)?

We can pull up it later when Velox backend gets to have similar demand.

Reply:
It seems  hard. Currently `GeneralOutIterator` is used in spark stages to get output both for Velox and CH.  Adding `cancel` only for CH `BatchIterator` will introduce a series of `if (iter.isInstanceOf[BatchIterator])`. And it it ugly.

Reply:
Sorry, I missed this message. It is possible to create resIter as BatchIterator instead of GeneralOutIterator in CH backend? In this way, we could use its custom member function directly.

> val resIter: GeneralOutIterator =
      transKernel.createKernelWithBatchIterator(
        inputPartition.plan,
        splitInfoByteArray,
        inBatchIters,
        false)

Reply:
> Sorry, I missed this message. It is possible to create resIter as BatchIterator instead of GeneralOutIterator in CH backend? In this way, we could use its custom member function directly.

+1

It's safe and ideal now to make `org.apache.gluten.vectorized.CHNativeExpressionEvaluator#createKernelWithBatchIterator` directly return a `BatchIterator` than `GeneralOutIterator`. We share `GeneralOutIterator`'s code only for legacy reason. It just includes some JNI method definition and doesn't have to be reused.

Reply:
Thanks for your advice. I'll try.